cirq-core/cirq/ops/eigen_gate.py

```diff
@@ -388,7 +388,7 @@ def _equal_up_to_global_phase_(self, other, atol):
             return False
 
         period = self_without_phase._period()
-        canonical_diff = (exponents[0] - exponents[1]) % period
+        canonical_diff = abs((exponents[0] - exponents[1])) % period
```

Comment:
Thank you for taking care of this!  The comparison seems to be still failing the https://github.com/quantumlib/Cirq/issues/5980#issue-1523071532 example for
```
gate1 = cirq.Z
gate2 = cirq.Z ** (3 - 1e-10)
```
The exponents are compared at a period=2, but the current formula fails for differences just below like 1.99999.

Please use [near_zero_mod](https://github.com/quantumlib/Cirq/blob/31bf1ac17fe21014b9c87ffe36a62a3a89f7c3b1/cirq-core/cirq/linalg/tolerance.py#L53) to compare the exponents instead.

Also please add unit tests for the exponent values where comparison fails before the fix.
I'd suggest to check `[1-eps, 1, 1+eps, 2-eps, 2, 2+eps, 3-eps, 3, 3+eps]` where the comparison should return True for exponents close to 1 and 3 and False for the exponents near 2.

Reply:
@siddharth-mehta - seems like this issue went off the radar.

Are you OK with me finishing this off or would you prefer to update the code yourself?